packages/api/internal/template-manager/client.go

```diff
@@ -34,9 +40,47 @@ func NewClient() (*GRPCClient, error) {
 		return nil, fmt.Errorf("failed to establish GRPC connection: %w", err)
 	}
 
-	client := template_manager.NewTemplateServiceClient(conn)
+	client := &GRPCClient{
+		Client:            template_manager.NewTemplateServiceClient(conn),
+		healthy:           template_manager.HealthState_Healthy,
+		lastHealthCheckAt: nil,
+		connection:        conn,
+	}
+
+	// periodically check for health status
+	go client.healthCheckSync(ctx)
 
-	return &GRPCClient{Client: client, connection: conn}, nil
+	return client, nil
+}
+
+func (a *GRPCClient) healthCheckSync(ctx context.Context) {
+	ticker := time.NewTicker(healthCheckInterval)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+			reqCtx, reqCtxCancel := context.WithTimeout(ctx, 2*time.Second)
+			healthStatus, err := a.Client.HealthStatus(reqCtx, nil)
+			healthCheckAt := time.Now()
```

Comment:
you can do this right here
```suggestion
			healthCheckAt := time.Now()
			a.lastHealthCheckAt = &healthCheckAt
```